### PR TITLE
Automatically fix up arc controller URL

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Fixes https://github.com/microsoft/azuredatastudio/issues/11137

@bergeron You see any issues with doing this? Just trying to help out the users (we do something similar for BDC) and I believe all the assumptions I've made are correct.

(also fixed up a small spelling mistake I noticed)

### CODE REVIEW

File: extensions/arc/src/ui/dialogs/connectControllerDialog.ts [hunk 1]
```diff
@@ -88,8 +88,21 @@ export class ConnectToControllerDialog {
 		if (!this.urlInputBox.value || !this.usernameInputBox.value || !this.passwordInputBox.value) {
 			return false;
 		}
+		let url = this.urlInputBox.value;
```
Comment (chlafreniere) [inline extensions/arc/src/ui/dialogs/connectControllerDialog.ts:91]: tests 😄

Comment (aaomidi) [reply extensions/arc/src/ui/dialogs/connectControllerDialog.ts:91]: This code seems interesting enough to maybe warrant it as a utility class to construct URLs from user input.

Comment (Charles-Gagnon) [reply extensions/arc/src/ui/dialogs/connectControllerDialog.ts:91]: Currently I doubt we'd ever need to use it outside this one particular place in this extension. I could potentially see this being a useful thing across multiple extensions but I'm not sure the use case is big enough to warrant the overhead of having to call through ext host/ext API. Do you see it being valuable elsewhere?

Comment (aaomidi) [reply extensions/arc/src/ui/dialogs/connectControllerDialog.ts:91]: What I meant was if it's a class it'd be easier to copy paste it to be used elsewhere if the need arises! But please don't block the PR on that :)

**Comment by Charles-Gagnon** [comment]
Merging this in now and will follow up with separate PR for tests

**Pull request merged**